session.rst

```diff
@@ -130,105 +391,303 @@ session metadata files:
+                    # - auth:
+                    #     - ['%env(REDIS_USER)%','%env(REDIS_PASSWORD)%']
 
     .. code-block:: xml
 
-        <!-- config/services.xml -->
         <?xml version="1.0" encoding="UTF-8" ?>
         <container xmlns="http://symfony.com/schema/dic/services"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd">
 
             <services>
-                <service id="session" class="Symfony\Component\HttpFoundation\Session\Session" public="true">
-                    <argument type="service" id="session.storage"/>
-                    <argument type="service" id="session.namespacedattributebag"/>
-                    <argument type="service" id="session.flash_bag"/>
+                <!-- you can also use \RedisArray, \RedisCluster or \Predis\Client classes -->
+                <service id="Redis" class="Redis">
+                    <call method="connect">
+                        <argument>%env(REDIS_HOST)%</argument>
+                        <argument>%env(int:REDIS_PORT)%</argument>
+                    </call>
+
+                    <!-- uncomment the following if your Redis server requires a password:
+                    <call method="auth">
+                        <argument>%env(REDIS_PASSWORD)%</argument>
+                    </call> -->
+
+                    <!-- uncomment the following if your Redis server requires a user and a password (when user is not default):
+                    <call method="auth">
+                        <argument>%env(REDIS_USER)%</argument>
+                        <argument>%env(REDIS_PASSWORD)%</argument>
+                    </call> -->
                 </service>
+            </services>
+        </container>
 
-                <service id="session.namespacedattributebag"
-                    class="Symfony\Component\HttpFoundation\Session\Attribute\NamespacedAttributeBag"
-                />
+    .. code-block:: php
+
+        // ...
+        $container
+            // you can also use \RedisArray, \RedisCluster or \Predis\Client classes
+            ->register('Redis', \Redis::class)
+            ->addMethodCall('connect', ['%env(REDIS_HOST)%', '%env(int:REDIS_PORT)%'])
+            // uncomment the following if your Redis server requires a password:
+            // ->addMethodCall('auth', ['%env(REDIS_PASSWORD)%'])
+            // uncomment the following if your Redis server requires a user and a password (when user is not default):
+            // ->addMethodCall('auth', ['%env(REDIS_USER)%', '%env(REDIS_PASSWORD)%'])
+        ;
+
+Now pass this ``\Redis`` connection as an argument of the service associated to the
+:class:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\RedisSessionHandler`.
+This argument can also be a ``\RedisArray``, ``\RedisCluster``, ``\Predis\Client``,
+and ``RedisProxy``:
```

Comment:
I suggest combining the previous and next code snippets into one, and move this to a code comment.